app/Commands/Compile.hs

```diff
@@ -0,0 +1,178 @@
+{-# LANGUAGE ApplicativeDo #-}
+{-# LANGUAGE TemplateHaskell #-}
+
+module Commands.Compile where
+
+import Commands.Extra
+import Data.ByteString qualified as BS
+import Data.FileEmbed qualified as FE
+import Data.Text.IO qualified as TIO
+import MiniJuvix.Prelude hiding (Doc)
+import Options.Applicative
+import System.Environment
+import System.Process qualified as P
+
+minijuvixBuildDir :: FilePath
+minijuvixBuildDir = ".minijuvix-build"
+
+data CompileTarget = TargetC | TargetWasm
+  deriving stock (Show)
+
+data CompileRuntime = RuntimeStandalone | RuntimeLibC
+  deriving stock (Show)
+
+data CompileOptions = CompileOptions
+  { _compileInputFile :: FilePath,
+    _compileTarget :: CompileTarget,
+    _compileRuntime :: CompileRuntime,
+    _compileOutputFile :: FilePath
+  }
+
+makeLenses ''CompileOptions
+
+parseCompile :: Parser CompileOptions
+parseCompile = do
+  _compileTarget <-
+    option
+      (eitherReader parseTarget)
+      ( long "target"
+          <> short 't'
+          <> metavar "TARGET"
+          <> value TargetWasm
+          <> showDefaultWith targetShow
+          <> help "select a target: wasm, c"
+      )
+  _compileRuntime <-
+    option
+      (eitherReader parseRuntime)
+      ( long "runtime"
+          <> short 'r'
+          <> metavar "RUNTIME"
+          <> value RuntimeStandalone
+          <> showDefaultWith runtimeShow
+          <> help "select a runtime: standalone, libc"
+      )
+  _compileOutputFile <-
+    option
+      str
+      ( long "output"
+          <> short 'o'
+          <> metavar "OUTPUT_FILE"
+          <> help "Path to output file"
+          <> action "file"
+          <> value "out"
+      )
+  _compileInputFile <- parserInputFile
+
+  pure CompileOptions {..}
+  where
+    parseTarget :: String -> Either String CompileTarget
+    parseTarget = \case
+      "wasm" -> Right TargetWasm
+      "c" -> Right TargetC
+      s -> Left $ "unrecognised target: " <> s
+
+    targetShow :: CompileTarget -> String
+    targetShow = \case
+      TargetC -> "c"
+      TargetWasm -> "wasm"
+
+    parseRuntime :: String -> Either String CompileRuntime
+    parseRuntime = \case
+      "standalone" -> Right RuntimeStandalone
+      "libc" -> Right RuntimeLibC
+      s -> Left $ "unrecognised runtime: " <> s
+
+    runtimeShow :: CompileRuntime -> String
+    runtimeShow = \case
+      RuntimeStandalone -> "standalone"
+      RuntimeLibC -> "libc"
+
+inputCFile :: CompileOptions -> FilePath
+inputCFile o = minijuvixBuildDir </> outputMiniCFile
+  where
+    outputMiniCFile :: FilePath
+    outputMiniCFile = takeBaseName (o ^. compileInputFile) <> ".c"
+
+runCompile :: CompileOptions -> Text -> IO (Either Text ())
```

Comment:
Do you think it would make sense to add an entry to the pipeline?
If so, we don't need to do it in this PR. I am only curious if you have thought about it.

Reply:
I can see the argument for adding to the pipeline. My reasoning for not adding it to the pipeline is that, as it currently stands, the compile command is just shelling-out to clang. i.e is external to the minijuvix compiler itself.